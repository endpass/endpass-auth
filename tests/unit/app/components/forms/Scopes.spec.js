import { shallowMount } from '@vue/test-utils';
import ScopesForm from '@/components/forms/Scopes';

describe('Scopes', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(ScopesForm);
  });

  describe('render', () => {
    it('should correctly render component', () => {
      expect(wrapper.name()).toBe('ScopesForm');
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render checkboxes with given scopes', () => {
      wrapper.setProps({
        scopes: ['foo', 'bar'],
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render error', () => {
      const error = 'error';

      wrapper.setProps({ error });

      expect(wrapper.find('[data-test=error-message]').text()).toBe(error);
    });

    it('should correctly disable submit button', () => {
      const submitButton = wrapper.find('[data-test=submit-button]');

      wrapper.setProps({
        loading: true,
      });

      expect(submitButton.text()).toBe('Loading...');
      expect(submitButton.attributes().disabled).toBeTruthy();

      wrapper.setProps({
        loading: false,
      });

      expect(submitButton.text()).toBe('Allow');
      expect(submitButton.attributes().disabled).toBeFalsy();

      wrapper.setProps({
        loading: false,
        scopes: ['foo', 'bar'],
      });
      wrapper.setData({ checkedScopes: [] });

      expect(submitButton.attributes().disabled).toBeTruthy();

      wrapper.setProps({
        loading: false,
        scopes: ['foo', 'bar'],
      });
      wrapper.setData({ checkedScopes: ['foo', 'bar'] });

      expect(submitButton.attributes().disabled).toBeFalsy();
    });
  });

  describe('behavior', () => {
    it('should submit form if all scopes are selected', () => {
      wrapper.setProps({
        loading: false,
        scopes: ['foo', 'bar'],
      });
      wrapper.setData({ checkedScopes: ['foo', 'bar'] });

      wrapper.find('form').trigger('submit');

      expect(wrapper.emitted().submit).toEqual([[['foo', 'bar']]]);
    });

    it('should not submit form if at least one scope is not selected', () => {
      wrapper.setProps({
        loading: false,
        scopes: ['foo', 'bar'],
      });
      wrapper.setData({ checkedScopes: [] });

      wrapper.find('form').trigger('submit');

      expect(wrapper.emitted().submit).toBeFalsy();
    });
  });
});
