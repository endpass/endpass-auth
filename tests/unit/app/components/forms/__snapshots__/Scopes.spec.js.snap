// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scopes render should correctly render component 1`] = `
<form>
  <form-field-stub>
    <message-stub>
      Application requests following scopes. Press "Allow" button to grant
      these permissions.
    </message-stub>
  </form-field-stub>
  <!---->
  <form-controls-stub>
    <v-button-stub type="primary" submit="true" fluid="true" data-test="submit-button">Allow</v-button-stub>
  </form-controls-stub>
</form>
`;

exports[`Scopes render should render checkboxes with given scopes 1`] = `
<form>
  <form-field-stub>
    <message-stub>
      Application requests following scopes. Press "Allow" button to grant
      these permissions.
    </message-stub>
  </form-field-stub>
  <!---->
  <form-field-stub>
    <v-checkbox-stub name="checkbox">
      foo
    </v-checkbox-stub>
  </form-field-stub>
  <form-field-stub>
    <v-checkbox-stub name="checkbox">
      bar
    </v-checkbox-stub>
  </form-field-stub>
  <form-controls-stub>
    <v-button-stub type="primary" submit="true" fluid="true" data-test="submit-button" disabled="true">Allow</v-button-stub>
  </form-controls-stub>
</form>
`;
